Tests/UnitTests/ImplementationDetailsTests.swift

```diff
@@ -0,0 +1,290 @@
+//
+// Copyright 2021 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+@testable import AEPEdge
+import XCTest
+
+class ImplementationDetailsTests: XCTestCase {
+    private let BASE_NAMESPACE = "https://ns.adobe.com/experience/mobilesdk/ios"
+    private let WRAPPER_REACT_NATIVE = "reactnative"
+    private let WRAPPER_CORDOVA = "cordova"
+    private let WRAPPER_FLUTTER = "flutter"
+    private let WRAPPER_UNITY = "unity"
+    private let WRAPPER_XAMARIN = "xamarin"
+
+    override func setUp() {
+        continueAfterFailure = false // fail so nil checks stop execution
+    }
+
+    // MARK: from tests
+
+    func testFrom_withVersion_withWrapperNone() {
+        let hubSharedState: [String: Any] = [
+            "version": "3.3.1",
+            "wrapper": ["type": "N"]
+        ]
+
+        guard let details = ImplementationDetails.from(hubSharedState) else {
+            XCTFail("ImplementationDetails returned nil when not expected.")
+            return
+        }
+
+        let actualResult = flattenDictionary(dict: details)
+
+        let expectedResult: [String: Any] = [
+            "name": "\(BASE_NAMESPACE)",
+            "environment": "app",
+            "version": "3.3.1+\(EdgeConstants.EXTENSION_VERSION)"
+        ]
+        assertEqual(expectedResult, actualResult)
+    }
+
+    func testFrom_withVersion_withWrapperReact() {
+        let hubSharedState: [String: Any] = [
+            "version": "3.3.1",
+            "wrapper": ["type": "R"]
+        ]
+
+        guard let details = ImplementationDetails.from(hubSharedState) else {
+            XCTFail("ImplementationDetails returned nil when not expected.")
+            return
+        }
+
+        let actualResult = flattenDictionary(dict: details)
+
+        let expectedResult: [String: Any] = [
+            "name": "\(BASE_NAMESPACE)/\(WRAPPER_REACT_NATIVE)",
+            "environment": "app",
+            "version": "3.3.1+\(EdgeConstants.EXTENSION_VERSION)"
+        ]
+        assertEqual(expectedResult, actualResult)
+    }
+
+    func testFrom_withVersion_withWrapperFlutter() {
```

Comment:
would be nice to have a test configuration for these repeating tests and switch the wrapperType val and expected namespace, would reduce this file lines significantly.